[Diagram] Post 이벤트 다이어그램 작성

codingbaraGo opened the pull request:
<!-- 제목은 `[도메인] PR 제목`으로 작성해주세요. (ex) [User] 소셜 로그인 필터 리팩토링 -->
<!-- PR의 타입은 Label을 통해 나타내주세요. -->

### 💻 작업 내용
<!-- 진행한 작업 내용에 대해 작성해주세요. -->
- 삭제, 목록 조회 다이어그램 추가
- 상세조회 캐시 버전 다이어그램 추가

### ✨ 리뷰 포인트
<!-- (ex) query 가 너무 복잡한 것 같은데 이 위주로 봐주세요. -->
게시글 상세조회의 경우 조회수관련 플로우를 이벤트로 처리하여 캐시에 반영하도록 만들었습니다.

### 📝 메모
<!-- PR 관련 전달사항이 있다면 이곳에 작성해주세요. -->
캐시 관련 플로우들도 설계하여 하나씩 추가해야 할 것 같네요.

### 🎯 관련 이슈
<!-- pr이 merge 되면 이슈가 자동으로 close 되도록 합니다. 만약 자동 close 를 하지 않고 이슈만 링크한다면 closed 키워드를 삭제해주세요.-->
#5

codingbaraGo committed:
docs(diagram): Post 이벤트 다이어그램 추가
- 삭제, 목록 조회 다이어그램 추가
- 상세조회 캐시 버전 다이어그램 추가

codingbaraGo committed:
chore(diagram): auto-generate SVGs [skip ci]

vwx12345 commented on document/puml/event/boardContext/postAggregate/deletePost.puml:
```diff
@@ -0,0 +1,47 @@
+end
+
+note right of PostController
+<b>Visitor</b>
```
공통 요소(Visitor) 는 한번만 작성하는게 좋을까요 아니면 사용하는 곳은 모두 표시하는게 좋을까요?

codingbaraGo replied:
한번만 작성하는걸로 하죠!
대표 엔티티에 작성하는걸로 해요

vwx12345 commented on document/puml/event/boardContext/postAggregate/deletePost.puml:
```diff
@@ -0,0 +1,47 @@
+'--- Controller
+PostController -> PostCommandService : deletePost(Visitor, postId, cmd)
+'--- Service
+PostCommandService -> PostRepository : findById(commentId)
```
commentId가 아닌 postId 여야 할 것 같습니다.

vwx12345 commented on document/puml/event/boardContext/postAggregate/deletePost.puml:
Post가 삭제될 때, Comment의 삭제는 안이루어지나요? 아니면 DomainEventPublisher 를 통해서 비동기로 진행하실 건가요?

codingbaraGo replied:
놓치고 있었던 부분이네요!
Post와 Comment 모두 soft delete이니 Comment를 deleted 처리하는것은 안될 듯 하고, Post만 soft delete 처리를 한 뒤, comment에 엑세스할 때 comment의 Post가 삭제된 상태인지 검증하는 로직이 필요하겠군요

codingbaraGo replied:
이 부분의 경우에는 isDeleted 컬럼을 commentStatus로 변경하도록 하겠습니다

codingbaraGo committed:
fix(diagram): Post 도메인 이벤트 다이어그램 수정
- 컨트롤러, 레포지토리에 Command/Query 명시
- getPostDetailCached.puml -> getPostDetail.puml 파일명 변경 및 다이어그램 내부 캐시 관련 정보 삭제

codingbaraGo committed:
[merge]: document/diagram-event-post <- develop
- 이벤트 다이어그램 디렉토리 구조 변경 적용

codingbaraGo committed:
fix(diagram): 이벤트 다이어그램 내부에서 Event handling 시퀸스 제거
